pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -625,6 +637,114 @@ func (pc *PodController) initializeSecondaryInterfaceStore() error {
 	return nil
 }
 
+// initializeSRIOVSecondaryInterfaceStore restores secondary interfaceStore for SR-IOV interfaces
+// when agent restarts. It will get the Pod info from the store of primary interfaces, and check
+// the NetworkStatus annotation of a Pod, then restore the SR-IOV interfaces based on the NetworkAttachmentDefinition
+// name and device's pci_address in the NetworkStatus.
+func (pc *PodController) initializeSRIOVSecondaryInterfaceStore() error {
+	knownInterfaces := pc.primaryInterfaceStore.GetInterfacesByType(interfacestore.ContainerInterface)
+	for _, ifconf := range knownInterfaces {
+		podNamespace := ifconf.ContainerInterfaceConfig.PodNamespace
+		podName := ifconf.ContainerInterfaceConfig.PodName
+		pod, err := pc.kubeClient.CoreV1().Pods(podNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
```

Comment:
This will introduce significant overhead in a scale cluster as it will trigger N (N=number of Pods) calls to kube-apiserver upon agent restart when SecondaryNetwork is enabled (which is by default since 2.4).

It should use the podInformer instead, but this initialization will need to be moved to `Run` as that's where the podInformer becomes synced.